common/thorhelper/thorsoapcall.cpp

```diff
@@ -50,6 +50,41 @@ using roxiemem::OwnedRoxieString;
 #define CONNECTION "Connection"
 
 unsigned soapTraceLevel = 1;
+StringBuffer soapSepString;
+static CriticalSection soapCrit;
+
+void setSoapSepString(StringBuffer &_soapSepString)
+{
+    CriticalBlock b(soapCrit);
+    soapSepString.clear().append(_soapSepString);
+}
+
+static void multiLineAppendReplace(StringBuffer &origStr, StringBuffer &newStr)
+{
```

Comment:
As an opt, we could index for '\r' and then append chunks between newlines.

Reply:
yes, if it was very time critical, might be worth two pass, so can allocate correct size once, reserve and use regular ptr's instead of calls to StringBuffer, but doubt worth it.